tests/test_hs_restclient.py

```diff
@@ -0,0 +1,166 @@
+""" Test hs_restclient functionality, with HydroShare REST end points mocked using httmock.
+
+    Note: tests must be run from 'tests' directory as in:
+
+        python test_hs_restclient.py
+"""
+import os
+import sys
+sys.path.append('../')
+import unittest
+from datetime import date, datetime
+import tempfile
+import shutil
+from zipfile import ZipFile
+import difflib
+import filecmp
+
+from httmock import with_httmock, HTTMock
+
+from hs_restclient import HydroShare
+import mocks.hydroshare
+
+
+class TestGetResourceTypes(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    @with_httmock(mocks.hydroshare.resourceTypes_get)
+    def test_get_resource_types(self):
+        hs = HydroShare()
+        res_type_proto = {'GenericResource',
+                          'ModelInstanceResource',
+                          'ModelProgramResource',
+                          'NetcdfResource',
+                          'RasterResource',
+                          'RefTimeSeries',
+                          'SWATModelInstanceResource',
+                          'TimeSeriesResource',
+                          'ToolResource'}
+
+        res_types = hs.getResourceTypes()
+        self.assertSetEqual(res_type_proto, res_types)
```

Comment:
so the tests for this break any time the backend adds a new resource type, which wouldn't break the API client at all?